test/js/cards/disk2.spec.ts

```diff
@@ -499,5 +503,196 @@ describe('DiskII', () => {
+            const nibble = diskII.ioSwitch(0x8c);  // read data
+            expect(nibble).toBe(0xFF);
+        });
+
+        it('reads several FF sync bytes', () => {
+            let cycles: number = 0;
+            mocked(mockApple2IO).cycles.mockImplementation(() => cycles);
+
+            const diskII = new DiskII(mockApple2IO, callbacks);
+            diskII.setBinary(1, 'DOS 3.3 System Master', 'woz', DOS33_SYSTEM_MASTER_IMAGE);
+
+            diskII.ioSwitch(0x89);  // turn on the motor
+            diskII.ioSwitch(0x8e);  // read mode
+
+            // The initial bytes in the image are: FF 3F CF F3
+            // making the bit stream:
+            //
+            //     1111 1111 0011 1111 1100 1111 1111 0011
+            //
+            // That's three FF sync bytes in a row. Assuming
+            // the sequencer is in state 2, each sync byte takes
+            // 32 clock cycles to read, is held for 8 clock
+            // cycles while the extra zeros are shifted in, then
+            // is held 8 more clock cycles while the sequencer
+            // reads the next two bits.  This means that 3 sync
+            // bytes will be available for 3 * 40 + 8 cycles.
+            for (let i = 0; i < 3 * 40 + 8; i++) {
+                cycles++;
+                const nibble = diskII.ioSwitch(0x8c);  // read data
+                if (nibble & 0x80) {
+                    // Nibbles are only valid when the high bit is set.
+                    // eslint-disable-next-line jest/no-conditional-expect
+                    expect(nibble).toBe(0xFF);
+                }
+            }
+        });
+
+        it('reads random garbage on uninitialized tracks', () => {
+            let cycles: number = 0;
+            mocked(mockApple2IO).cycles.mockImplementation(() => cycles);
+
+            const diskII = new DiskII(mockApple2IO, callbacks);
+            diskII.setBinary(1, 'DOS 3.3 System Master', 'woz', DOS33_SYSTEM_MASTER_IMAGE);
+
+            // Step to track 0.5
+            diskII.ioSwitch(0x89);  // turn on the motor
+            diskII.ioSwitch(0x81);  // coil 0 on
+            diskII.ioSwitch(0x83);  // coil 1 on
+            diskII.ioSwitch(0x80);  // coil 0 off
+            diskII.ioSwitch(0x82);  // coil 1 off
+            diskII.ioSwitch(0x8e);  // read mode
+
+            // Read 5 nibbles
+            const nibbles: byte[] = [];
+            let read = false;
+            while (nibbles.length < 5) {
+                cycles++;
+                const nibble = diskII.ioSwitch(0x8c);  // read data
+                const qa = nibble & 0x80;
+                if (qa && !read) {
+                    nibbles.push(nibble);
+                    read = true;
+                }
+                if (!qa && read) {
+                    read = false;
+                }
+            }
+            // Test that the first doesn't equal any of the others.
+            // (Yes, this test could fail with some bad luck.)
```

Comment:
Hmm. This did wind up failing here. I wonder if you could compute deviation or some other measure of value distribution.
https://github.com/whscullin/apple2js/runs/7084555004?check_suite_focus=true